### PR TITLE
Add in support for casting binary to string

### PR DESCRIPTION
**Pull request by revans2**
This is needed to be able to accelerate zordering in delta lake open source.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala [hunk 1]
```diff
@@ -1367,7 +1367,7 @@ class CastChecks extends ExprChecks {
   val sparkStringSig: TypeSig = cpuNumeric + BOOLEAN + TIMESTAMP + DATE + CALENDAR + STRING +
       BINARY + GpuTypeShims.additionalTypesStringCanCastTo
 
-  val binaryChecks: TypeSig = BINARY
+  val binaryChecks: TypeSig = STRING + BINARY
```
Comment (abellina) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala:1370]: A comment here explaining why binary can be STRING would be nice.

Comment (revans2) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala:1370]: I am confused by this. Spark supports the cast. Why do we need to explain that.

**Comment by revans2** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala (hunk 1 above)
Comment (abellina) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala:1370]: sorry, I misunderstood this.

**Review by abellina — APPROVED** [review]

**Pull request merged**